src/hypergraph/runners/_shared/validation.py

```
@@ -175,21 +175,38 @@ def validate_runner_compatibility(
     Raises:
         GraphConfigError: If graph contains features incompatible with map
     """
-    # Phase 2: Check for interrupt nodes
-    # For now, all graphs are map-compatible
-    pass
+    if graph.has_interrupts:
+        raise IncompatibleRunnerError(
+            "Graph contains InterruptNode(s) which are incompatible with .map(). "
+            "Use .run() for graphs with interrupts.",
+            capability="supports_interrupts",
+        )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)